From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alistair Strachan <astrachan@google.com>
Date: Wed, 24 Oct 2018 13:58:11 -0700
Subject: ANDROID: Removed check for asm-goto

Cherry pick was manually applied due to conflicts with
upstream commits: e9666d10a56 and 829fe4aa9ac
--------
Revert "x86: Force asm-goto"

This reverts commit e501ce957a786ecd076ea0cfb10b114e6e4d0f40.

This change broke building the x86_64 kernel with clang. The kernel
still builds and works fine without asm-goto support. Revert this
change to unblock testing clang kernels on cuttlefish.

Bug: 118142806
Bug: 120440614
Bug: 132629930
Change-Id: Ib32498acc0596264f8cd15de0b603579dd643dd7
Signed-off-by: Alistair Strachan <astrachan@google.com>
Signed-off-by: Ram Muthiah <rammuthiah@google.com>
---
 arch/x86/Makefile | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/arch/x86/Makefile b/arch/x86/Makefile
index 94df0868804b..408bb5532d1b 100644
--- a/arch/x86/Makefile
+++ b/arch/x86/Makefile
@@ -298,10 +298,6 @@ vdso_install:
 
 archprepare: checkbin
 checkbin:
-ifndef CONFIG_CC_HAS_ASM_GOTO
-	@echo Compiler lacks asm-goto support.
-	@exit 1
-endif
 ifdef CONFIG_RETPOLINE
 ifeq ($(RETPOLINE_CFLAGS),)
 	@echo "You are building kernel with non-retpoline compiler." >&2
